src/main/kotlin/com/sourcegraph/cody/listeners/CodyFocusChangeListener.kt

```diff
@@ -15,8 +16,15 @@ class CodyFocusChangeListener(val project: Project) : FocusChangeListener {
       EditorChangesBus.documentChanged(project, textDocument)
       CodyAgentService.withAgent(project) { agent: CodyAgent ->
         agent.server.textDocumentDidFocus(textDocument)
+        agent.server.window_didChangeFocus(Window_DidChangeFocusParams(true))
       }
       IgnoreOracle.getInstance(project).focusedFileDidChange(textDocument.uri)
     }
   }
+
+  override fun focusLost(editor: Editor) {
```

Comment:
Switching b/w files in the project calls `focusLost` and `focusGained` immediately afterwards.

Is it ok?

Reply:
I think that is incorrect implementation.

What you need is new listener:

```
    val frame = WindowManager.getInstance().getFrame(project)
    val listener =
        object : WindowAdapter() {
          override fun windowActivated(e: WindowEvent?) {
            super.windowActivated(e)
            ApplicationManager.getApplication().executeOnPooledThread { getAuthenticationState() }
          }
        }
    frame?.addWindowListener(listener)
    Disposer.register(this) { frame?.removeWindowListener(listener) }
```

(this code is copied, please take a look at `CodyAuthenticationManager::addAuthChangeListener`)

Reply:
done